File: api/src/main/java/net/blueshell/api/mapper/SocialMapper.java
package net.blueshell.api.mapper;

import net.blueshell.api.common.enums.PlatformType;
import net.blueshell.api.dto.SocialDTO;
import net.blueshell.api.model.Blog;
import org.mapstruct.*;
import org.springframework.beans.factory.annotation.Value;

@Mapper(componentModel = "spring", nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
public abstract class SocialMapper {

    @Value("${frontend.url}")
    private String frontendUrl;

    @BeanMapping(ignoreByDefault = true)
    public abstract SocialDTO toSocialDTO(Blog blog);

    @AfterMapping
    protected void afterToSocialDTO(Blog blog, @MappingTarget SocialDTO dto) {
        dto.setUrl(frontendUrl + "/blogs" + blog.getId());
        dto.setTitle(blog.getTitle());
        dto.setText(dto.getText());
        PlatformType[] platforms = {PlatformType.FACEBOOK, PlatformType.TWITTER, PlatformType.INSTAGRAM, PlatformType.LINKEDIN};
        dto.setPlatforms(platforms);
    }
}

File: api/src/main/kotlin/net/blueshell/api/mapper/SocialMapper.kt
package net.blueshell.api.mapper

import net.blueshell.api.common.enums.PlatformType
import net.blueshell.api.dto.SocialDTO
import net.blueshell.api.model.Blog
import org.mapstruct.AfterMapping
import org.mapstruct.BeanMapping
import org.mapstruct.Mapper
import org.mapstruct.MappingTarget
import org.springframework.beans.factory.annotation.Value

@Mapper(componentModel = "spring")
abstract class SocialMapper {
    @Value("\${frontend.url}")
    private val frontendUrl: String = null

    @BeanMapping(ignoreByDefault = true)
    abstract fun toSocialDTO(blog: Blog): SocialDTO

    @AfterMapping
    protected fun afterToSocialDTO(blog: Blog, @MappingTarget dto: SocialDTO) {
        dto.setUrl(frontendUrl + "/blogs" + blog.getId())
        dto.setTitle(blog.getTitle())
        dto.setText(dto.getText())
        val platforms = arrayOf<PlatformType>(
            PlatformType.FACEBOOK,
            PlatformType.TWITTER,
            PlatformType.INSTAGRAM,
            PlatformType.LINKEDIN
        )
        dto.setPlatforms(platforms)
    }
}
